x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/NlpTokenizer.java

```diff
@@ -10,43 +10,217 @@
 import org.elasticsearch.core.Releasable;
 import org.elasticsearch.xpack.core.ml.inference.trainedmodel.BertTokenization;
 import org.elasticsearch.xpack.core.ml.inference.trainedmodel.MPNetTokenization;
+import org.elasticsearch.xpack.core.ml.inference.trainedmodel.RobertaTokenization;
 import org.elasticsearch.xpack.core.ml.inference.trainedmodel.Tokenization;
 import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
 import org.elasticsearch.xpack.ml.inference.nlp.NlpTask;
 import org.elasticsearch.xpack.ml.inference.nlp.Vocabulary;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 import java.util.OptionalInt;
+import java.util.stream.Collectors;
 
 import static org.elasticsearch.xpack.core.ml.inference.trainedmodel.NlpConfig.TOKENIZATION;
 import static org.elasticsearch.xpack.core.ml.inference.trainedmodel.NlpConfig.VOCABULARY;
 
 /**
  * Base tokenization class for NLP models
  */
-public interface NlpTokenizer extends Releasable {
+public abstract class NlpTokenizer implements Releasable {
+    abstract int clsTokenId();
 
-    TokenizationResult buildTokenizationResult(List<TokenizationResult.Tokens> tokenizations);
+    abstract int sepTokenId();
 
-    List<TokenizationResult.Tokens> tokenize(String seq, Tokenization.Truncate truncate, int span, int sequenceId);
+    abstract int maxSequenceLength();
 
-    TokenizationResult.Tokens tokenize(String seq1, String seq2, Tokenization.Truncate truncate, int sequenceId);
+    abstract boolean isWithSpecialTokens();
 
-    NlpTask.RequestBuilder requestBuilder();
+    abstract int getNumExtraTokensForSeqPair();
 
-    OptionalInt getPadTokenId();
+    public abstract TokenizationResult buildTokenizationResult(List<TokenizationResult.Tokens> tokenizations);
 
-    String getPadToken();
+    /**
+     * Tokenize the input according to the basic tokenization
+     * options then perform the configured tokenization with the given vocabulary.
+     *
+     * The result is the tokens ids, a map of the
+     * token position to the position of the token in the source for
+     * each input string grouped into a {@link Tokenization}.
+     *
+     * @param seq Text to tokenize
+     * @return A list of {@link Tokenization}
+     */
+    public List<TokenizationResult.Tokens> tokenize(String seq, Tokenization.Truncate truncate, int span, int sequenceId) {
+        var innerResult = innerTokenize(seq);
+        List<? extends DelimitedToken.Encoded> wordPieceTokenIds = innerResult.tokens();
```

Comment:
Yeah, I need to rename the variables as they may not be "wordpiece" tokens. I will do that soon.